core/src/main/java/org/mapstruct/EnumMapping.java

```diff
@@ -111,13 +111,22 @@
      *
      * @return the name transformation strategy
      */
-    String nameTransformationStrategy();
+    String nameTransformationStrategy() default "";
 
     /**
      * The configuration that should be passed on the appropriate name transformation strategy.
      * e.g. a suffix that should be applied to the source enum when doing name based mapping.
      *
      * @return the configuration to use
      */
-    String configuration();
+    String configuration() default "";
+
+    /**
+     * Exception that should be thrown by the generated code if no mapping matches.
```

Comment:
we probably should mention that:
1. The exception should have an empty constructor
2. The exception (when checked) should be added to the method signature.

Reply:
I'll add that.

Reply:
great.. After that, you could merge the PR..